<a href="https://colab.research.google.com/github/cesarrivera52/Entendimientodedatos_1erSem/blob/main/Fase_2_Entendimiento_de_los_datos_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

César Alejandro Rivera Guzmán

A01567012

German Dominguez

Matemáticas y ciencia de datos para la toma de decisiones

Fase 2
Parte 1


In [ ]:
import pandas as pd
datos_consumo = pd.read_excel('A01567012_Actividad5_Registro (1).xlsx')


In [ ]:
datos_consumo.head()



Fecha (dd/mm/aa)  ...                                Unnamed: 11
0       2021-08-16  ...                                        NaN
1       2021-08-16  ...                                  Tu nombre
2       2021-08-16  ...                                        NaN
3       2021-08-16  ...  Indicar el momento de consumo en la tabla
4       2021-08-17  ...                                   Desayuno

[5 rows x 12 columns]

In [ ]:
datos_consumo.shape


(194, 12)

In [ ]:
datos_consumo.columns


Index(['Fecha (dd/mm/aa)', 'Momento', 'Nombre alimento', 'Calorías (kcal)',
       'Carbohidratos (g)', 'Lípidos/grasas (g)', 'Proteína (g)', 'Sodio (mg)',
       'https://www.fatsecret.com.mx/calor%C3%ADas-nutrici%C3%B3n/',
       'Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11'],
      dtype='object')

In [ ]:
datos_consumo.dtypes

Fecha (dd/mm/aa)                                              datetime64[ns]
Momento                                                               object
Nombre alimento                                                       object
Calorías (kcal)                                                      float64
Carbohidratos (g)                                                    float64
Lípidos/grasas (g)                                                   float64
Proteína (g)                                                         float64
Sodio (mg)                                                           float64
https://www.fatsecret.com.mx/calor%C3%ADas-nutrici%C3%B3n/            object
Unnamed: 9                                                           float64
Unnamed: 10                                                           object
Unnamed: 11                                                           object
dtype: object

# **Procedimiento(Parte 1, Programa)**:
Para poder realizar este programa lo que se tuvo que realizar fue lo siguiente, primero se tuvo que trabajar en un excel, donde se tienen que ingresar día con día lo que se consume, ingresar las calorías, proteínas, carbohidratos, lípidos, etc. Lo segundo que se hizo fue subir el archivo de excel donde estan los datos a Google Colab. Por último, se crea una variable que muestre el archivo de excel, y por último escribir lo que se desea mostrar, por ejemplo, lo primero que se ingreso en la tabla, la cantidad de filas y tablas que tienen, columnas, tipo de datos, etc.



Entendimiento de los datos:

Para poder realizar un programa de ciencia de datos, tenemos que tener muy presente que tipos de datos se pueden encontrar, ya que cada dato que se nos da, es diferente y te puede ayudar para poder entender o llegar a diferentes valores. Como se menciona en la lectura existen difernetes datos que provienen de la variedad de fuentes, como los siguientes:
Datos Existentes: Son datos que incluyen una variedad de datos, como encuestas o páginas web.
Datos adquiridos: Son datos que tienen la capacidad de completarse con información externa, como los datos demográficos, datos de la competencia, etc.
Datos adicionales: Los datos adicionales son aquellos que cuando las fuentes anteriores no son lo suficiente para llenar los datos necesarios, estas se pueden utilizar para acompletar y dejar claro con más información, estos datos son las redes sociales, artículos, etc. 

Para clasificar los tipos de datos normalmente se pueden hacer preguntas para saber que son, las preguntas para saber que tipo son pueden ser las siguientes:
"¿Cuáles son tus datos existentes, datos adquiridos y datos adicionales?, ¿Qué tipos de datos se analizarán?, ¿Qué atributos (columnas) de la base de datos parecen más prometedores?,  ¿Qué atributos parecen irrelevantes y pueden ser excluidos?, ¿Cuál es la relación de los datos y la hipótesis del proyecto?" (Anonimo, s.f.),  etc.

Aprende sobre: Entendimiento y clasificación de los datos (Fase 2). (s. f.). Canvas. Recuperado 10 de octubre de 2021, de https://experiencia21.tec.mx/courses/198531/pages/aprende-sobre-entendimiento-y-clasificacion-de-los-datos-fase-2?module_item_id=9163147



¿Cuáles son tus datos existentes, datos adquiridos y datos adicionales?

Mis datos existentes son mi peso, edad, y mi altura.
Los datos adquiridos son los datos que agregue extra, las calorías, proteínas, carbohidratos, lípidos, para obtener estos datos, los busque en una página web que me arrojaba los datos que necesitaba para poder hacer una calculadora de calorías y el IMC.
Los datos adicionales serían la fecha de consumo, el tipo de comida, etc.

• ¿Qué tipos de datos se analizarán?

Los datos que se estan analizando son datos correlacionales, ya que dependen cada uno para poder poner en práctica los diferentes programas, como el IMC y la calculadora de calorías.

• ¿Qué atributos (columnas) de la base de datos parecen más prometedores?
Para mí, todos los datos que se tienen en la base de datos son importantes.
De las 4 variables con los nutrientes ¿cuáles son útiles para el análisis?
Las variables de carbohidratos, proteínas y lípidos, ya que son los más importantes para nuestra calculadora.

• ¿Qué atributos parecen irrelevantes y pueden ser excluidos?
La columna de sodio, ya que no se utiliza para nada en la formula que se utiliza.

• ¿Hay datos suficientes (filas) para sacar conclusiones generalizables o hacer predicciones precisas?
Si, la base de datos cuenta con 194 columnas de información.

• ¿Hay demasiados atributos para realizar un modelo que sea fácil de interpretar?
La mayoría de los datos con los que se cuenta pueden utilizarse, ya que cuando se realizaron las gráficas para validar el modelo, estas si cumplían los lineamientos requeridos.

• ¿De dónde se obtuvieron los datos? 
Los datos fueron obtenidos primero de mi, ya que yo era el que ponía el nombre del alimento que consumía, después de eso se buscaba en la página que se tiene para ver la cantidad de nutrientes y por ultimo llenarla. ¿Se están fusionando varias fuentes de datos? Si, ¿hay áreas que podrían plantear un problema al fusionar?
Se utilizan las diferentes variables como se menciono anteriormente para crear u obtener los resultados de los IMC y las calculadoras de calorías.

• ¿Hay algún plan para manejar los valores faltantes en cada una de las fuentes de datos?
Para las variables que no cuenten. on algún datos, se planea que se elimine, para poder tener una fuente de datos más organizada.

• ¿Cuántos datos están accesibles o disponibles y cómo está la calidad de los mismos?.
La cantidad de filas para los datos accesibles son 5, y columnas 193.

• ¿Cuál es la relación de los datos y la hipótesis del proyecto?
La cantidad de alimento que se consume, y la cantidad de nutrientes y propiedades que tiene cada aliemnto, son las principales variables que afentan a los programas de Python ya antes creados que son para el resultado del IMC que posteriormente tendras, y con las calorías que contiene cada comida.